src/libtiled/mapobject.cpp

```diff
@@ -428,6 +428,16 @@ void MapObject::syncWithTemplate()
 
     if (!propertyChanged(MapObject::VisibleProperty))
         setVisible(base->isVisible());
+
+    QMapIterator<QString, Properties> it(base->components());
+    while (it.hasNext()) {
+        it.next();
+        if (!hasComponent(it.key())) {
+            addComponent(it.key(), Properties {});
+            Properties& properties = componentProperties(it.key());
+            Tiled::mergeProperties(properties, it.value());
```

Comment:
This is not very different from what the `mergeComponents` method was doing before, right?

What I meant to say is, we shouldn't actively copy down properties from the template to the instance, because the existence of the property on the instance means it overrides the value from the template. At least, that's what it currently means for the properties, and I think it would be a good idea to use the same approach with the components.

We'd need something like the `Object::resolvedProperties`, but then `Object::resolvedComponents`, which returns the components of the object, including inherited components and the values of their properties. That's what we display in the Properties view (grayed out values, to indicate when they are inherited vs. when they are overridden).

Reply:
No, previously it copied from template to instance no questions asked.
Now, it's copying only if the instance does not have the component. (iterates the components).

I didn't plan on adding properties inheritance for components, I'm not sure how much work is that...

Reply:
Ok, then it's fine if this is making an initial copy for now. It can be removed when we do add support for inheritance to components in the future. The code can then be simplified to:
```c++
        if (!hasComponent(it.key()))
            addComponent(it.key(), it.value());
```